python/grass/jupyter/baseseriesmap.py

```diff
@@ -0,0 +1,197 @@
+        self, width=None, height=None, env=None, use_region=False, saved_region=None
+    ):
+        """Creates an instance of the visualizations class.
+
+        :param int width: width of map in pixels
+        :param int height: height of map in pixels
+        :param str env: environment
+        :param use_region: if True, use either current or provided saved region,
+                          else derive region from rendered layers
+        :param saved_region: if name of saved_region is provided,
+                            this region is then used for rendering
+        """
+
+        # Copy Environment
+        if env:
+            self._env = env.copy()
+        else:
+            self._env = os.environ.copy()
+
+        self.baseseries = None
+        self._base_layer_calls = []
+        self._base_calls = []
+        self._baseseries_added = False
+        self._layers_rendered = False
+        self._base_filename_dict = {}
+        self._width = width
+        self._height = height
+
+        self._slider_description = ""
+        self._labels = []
+        self._indices = []
+
+        # Create a temporary directory for our PNG images
+        # Resource managed by weakref.finalize.
+        self._tmpdir = (
+            # pylint: disable=consider-using-with
+            tempfile.TemporaryDirectory()
+        )
+
+        def cleanup(tmpdir):
+            tmpdir.cleanup()
+
+        weakref.finalize(self, cleanup, self._tmpdir)
+
+        # Handle regions in respective classes
+        self.use_region = use_region
+        self.saved_region = saved_region
+
+    def __getattr__(self, name):
+        """
+        Parse attribute to GRASS display module. Attribute should be in
+        the form 'd_module_name'. For example, 'd.rast' is called with 'd_rast'.
+        """
+        # Check to make sure format is correct
+        if not name.startswith("d_"):
+            raise AttributeError(_("Module must begin with 'd_'"))
+        # Reformat string
+        grass_module = name.replace("_", ".")
+        # Assert module exists
+        if not shutil.which(grass_module):
+            raise AttributeError(_("Cannot find GRASS module {}").format(grass_module))
+        # if this function is called, the images need to be rendered again
+        self._layers_rendered = False
+
+        def wrapper(**kwargs):
+            if not self._baseseries_added:
+                self._base_layer_calls.append((grass_module, kwargs))
+            else:
+                if self._base_calls is not None:
+                    for row in self._base_calls:
+                        row.append((grass_module, kwargs))
+                else:
+                    self._base_calls.append((grass_module, kwargs))
+
+        return wrapper
+
+    def _render_baselayers(self, img):
+        """Add collected baselayers to Map instance"""
+        for grass_module, kwargs in self._base_layer_calls:
+            img.run(grass_module, **kwargs)
+
+    def render(self):
```

Comment:
The name of this method probably should not be `render`, since the render method in the subclasses are not meant to override this, but they use it in their render methods. So either `_render` or a name that describes what this method actually does. Make sure the documentation of the method actually matches what the method is doing.